torch/_decomp/decompositions.py

```diff
@@ -1370,7 +1378,127 @@ def native_batch_norm(
     if input.device.type == "cpu":
         save_mean = save_mean.to(dtype=input.dtype)
         save_rstd = save_rstd.to(dtype=input.dtype)
-    return output.to(dtype=input.dtype), save_mean, save_rstd
+    return (
```

Comment:
nit: the 2 extra returns are marked optional, but you always return them.

Reply:
Doesn't the optional mean that I return None instead of a Tensor sometimes? (Which is true)

Reply:
It looks like in the case when `training=False,` we always set new_running_mean/new_running_var to be the current running_mean/running_var, and it's possible for `train` to be False, but for the user to still pass in a non-None running_mean/running_var.

This is fine to clean up later though: maybe ensuring that we always return None, and then in the non-functional decomps that call this helper, asserting that the last two outputs are None.